src/app.controller.ts

```diff
@@ -597,4 +601,49 @@ export class AppController {
     };
   }
 
+  @Get('user-edit')
+  @UseFilters(UnauthorizedExceptionFilter)
+  @Render('ep-dashboard-edit')
+  @UseGuards(JwtAuthGuard)  
+  async userEdit(
+    @Req() req,
+    @Query('msg') msg: string
+  ): Promise<any> {
+    req.app.locals.layout = 'ep-main';
+    return {
+      messages: msg && msg.split(','),
+      timezones: Utils.getTimezones(),
+      countryCodes: Utils.getCountryCodes(),
+      user: req.user,
+      environment: process.env.ENVIRONMENT,
+      gtag: process.env.GOOGLE_TAG,
+    };
+  }  
+
+  @Post('user-edit')
+  @UseFilters(UnauthorizedExceptionFilter)
+  @UseFilters(UserEditBadRequestExceptionFilter)
+  @UseGuards(JwtAuthGuard)  
+  async updateUser(
+    @Req() req,
+    @Body() body: UpdateUserUIDto,
+    @Res() res: Response
+  ): Promise<any> {
+    const user = await this.userService.findByUserId(req.user.userid);
+    if(!user)
+      throw new BadRequestException("user not found!");
+
+    const userWithEmail = await this.userService.findByEmail(body.email);
+    if(userWithEmail && (userWithEmail.userid.toString() != user.userid.toString()))
+      throw new BadRequestException("This E-Mail is being used by another user!");
+    
+    const userToUpdate = plainToClassFromExist(user,body);
+    userToUpdate.emails.push(body.email);
```

Comment:
Could we also save this into a platform connection object? This would make the logic of data gathered from earth points consistent with data such as emails from other platforms.

The call to create the first platform connection:

```
    await this.platformConnectionService.create({
      userid,
      profile_id: userid,
      platform: 'earthpoints',
      auth_token: null,
      auth_expiration: null,
      emails: [body.email],
    });
```

If a platform connection object already exists for earthpoints we would need to retrieve it and update it instead.